tests/TestCase.php

```diff
@@ -2,8 +2,28 @@
 
 namespace WpOrg\Requests\Tests;
 
+use WpOrg\Requests\Requests;
 use Yoast\PHPUnitPolyfills\TestCases\TestCase as Polyfill_TestCase;
 
 class TestCase extends Polyfill_TestCase {
 
+	/**
+	 * Data provider for use in tests which need to be run against all default supported transports.
+	 *
+	 * @var array
+	 */
+	public function transportProvider() {
+		static $data;
+
+		if (!isset($data)) {
+			$data = array();
+
+			foreach (Requests::DEFAULT_TRANSPORTS as $transport) {
+				$name        = substr($transport, (strrpos($transport, '\\') + 1));
+				$data[$name] = array($transport);
+			}
+		}
+
+		return $data;
```

Comment:
Given that tests are not run in production and thus performance is not that big of a concern, I'd rather keep static state out of tests whenever possible, to avoid potentially having flaky tests because of it.

```suggestion
		$data = array();

		foreach (Requests::DEFAULT_TRANSPORTS as $transport) {
			$name        = substr($transport, (strrpos($transport, '\\') + 1));
			$data[$name] = array($transport);
		}

		return $data;
```